subprojects/core/src/main/java/org/gradle/api/internal/DefaultDomainObjectCollection.java

```diff
@@ -38,16 +38,13 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Iterator;
-import java.util.List;
-import java.util.ListIterator;
 
 public class DefaultDomainObjectCollection<T> extends AbstractCollection<T> implements DomainObjectCollection<T>, WithEstimatedSize {
```

Comment:
It would be good to see some unit test coverage for the new behaviour of this type plus the other collection types. There's a couple of abstract test classes (`AbstractDomainObjectCollectionSpec` and `AbstractNamedDomainObjectCollectionSpec` that can be used to add some contract tests).

Reply:
Done.